File: Algorithms/Sorting/insertion_sort.c
#include <stdio.h>

void
insertion_sort (int* seq, int n)
{
  for (int i = 0; i < n; ++i)
    {
      int cur = seq[i];
      int j = i - 1;
      while (j >= 0 && seq[j] > cur)
	seq[j + 1] = seq[j--];
      seq[j + 1] = cur;
    }
}
  

int
main ()
{
  int seq[] = {3, 5, 2, 4, 1};
  int n = 5;

  insertion_sort (seq, n);

  for (int i = 0; i < n; ++i)
    printf ("%d ", seq[i]);
  printf ("\n");
}

File: Algorithms/Sorting/insertion_sort.go
package main

import "fmt"

func insertion_sort(seq []int) {
  for i := 0; i < len(seq); i++ {
    cur := seq[i]
    j := i - 1
    for j >= 0 && seq[j] > cur {
      seq[j + 1] = seq[j];
      j--
    }
    seq[j + 1] = cur
  }
}

func main() {
  seq := []int{3, 5, 2, 4, 1}

  insertion_sort(seq)
  fmt.Println(seq)
}
